crates/sui-framework/sources/governance/genesis.move

```diff
@@ -4,128 +4,206 @@
-            let network_pubkey = *vector::borrow(&validator_network_pubkeys, i);
-            let worker_pubkey = *vector::borrow(&validator_worker_pubkeys, i);
-            let proof_of_possession = *vector::borrow(&validator_proof_of_possessions, i);
-            let name = *vector::borrow(&validator_names, i);
-            let description = *vector::borrow(&validator_descriptions, i);
-            let image_url = *vector::borrow(&validator_image_urls, i);
-            let project_url = *vector::borrow(&validator_project_urls, i);
-            let net_address = *vector::borrow(&validator_net_addresses, i);
-            let p2p_address = *vector::borrow(&validator_p2p_addresses, i);
-            let primary_address = *vector::borrow(&validator_primary_addresses, i);
-            let worker_address = *vector::borrow(&validator_worker_addresses, i);
-            let gas_price = *vector::borrow(&validator_gas_prices, i);
-            let commission_rate = *vector::borrow(&validator_commission_rates, i);
-            vector::push_back(&mut validators, validator::new(
+            let GenesisValidatorMetadata {
+                name,
+                description,
+                image_url,
+                project_url,
+                sui_address,
+                gas_price,
+                commission_rate,
+                protocol_public_key,
+                proof_of_possession,
+                network_public_key,
+                worker_public_key,
+                network_address,
+                p2p_address,
+                primary_address,
+                worker_address,
+            } = *vector::borrow(&genesis_validators, i);
+
+            let validator = validator::new(
                 sui_address,
-                pubkey,
-                network_pubkey,
-                worker_pubkey,
+                protocol_public_key,
+                network_public_key,
+                worker_public_key,
                 proof_of_possession,
                 name,
                 description,
                 image_url,
                 project_url,
-                net_address,
+                network_address,
                 p2p_address,
                 primary_address,
                 worker_address,
-                // Initialize all validators with uniform stake taken from the subsidy fund.
-                // TODO: change this back to take from subsidy fund instead.
-                option::some(balance::split(&mut sui_supply, initial_validator_stake_mist)),
+                // Initialize all validators with no stake. Stake distribution
+                // will be handled by the 'allocate_tokens' call bellow.
+                option::none(),
                 gas_price,
                 commission_rate,
-                true, // validator is active right away
                 ctx
-            ));
+            );
+
+            vector::push_back(&mut validators, validator);
```

Comment:
I realized we aren't checking duplicates here.. is this fine?

Reply:
We weren't checking duplicates before, happy to add that in as a follow up (since there's still more work to get in after this lands)

Reply:
Pushed commit to check for duplicates.

Reply:
So actually, i don't think multiple allocations to the same address would be an issue. It would just create an extra sui object (which may be useful). We'll also expect multiple allocations to a single address for staking purposes anyway.